### PR TITLE
python310Packages.simpleitk: init at 2.2.1

### PR DESCRIPTION
**Pull request by bcdarwin**
###### Description of changes

Add a package.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

@cfhammill

### CODE REVIEW

File: pkgs/development/libraries/simpleitk/default.nix [hunk 1]
```diff
@@ -4,12 +4,10 @@ stdenv.mkDerivation rec {
   pname = "simpleitk";
   version = "2.2.1";
 
-  outputs = [ "out" "dev" ];
-
   src = fetchFromGitHub {
     owner = "SimpleITK";
     repo = "SimpleITK";
-    rev = "v${version}";
+    rev = "refs/tags/v${version}";
```
Comment (OmnipotentEntity) [inline pkgs/development/libraries/simpleitk/default.nix:10]: Why was this changed exactly?  `"v${version}"` ought to work fine, and is more customary across nixpkgs to my knowledge.

Comment (bcdarwin) [reply pkgs/development/libraries/simpleitk/default.nix:10]: However, I haven't looked for a reference in the Nix manual for this.

Comment (OmnipotentEntity) [reply pkgs/development/libraries/simpleitk/default.nix:10]: It looks like it's originally from this https://discourse.nixos.org/t/why-is-the-r-ryantm-bot-using-refs-tags-in-the-version/20519  And it has an entry here: https://nixos.org/manual/nixpkgs/stable/#fetchgit

Thanks for alerting me to this.

File: pkgs/development/libraries/simpleitk/default.nix [hunk 2]
```diff
@@ -4,12 +4,10 @@ stdenv.mkDerivation rec {
   pname = "simpleitk";
   version = "2.2.1";
 
-  outputs = [ "out" "dev" ];
-
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/simpleitk/default.nix:8]: This means more likely the cake file is wrong

File: pkgs/development/libraries/itk/generic.nix [hunk 3]
```diff
@@ -21,6 +21,13 @@ stdenv.mkDerivation rec {
     hash = "sha256-deJbza36c0Ohf9oKpO2T4po37pkyI+2wCSeGL4r17Go=";
   };
 
+  itkSimpleITKFiltersSrc = fetchFromGitHub {
+    owner = "InsightSoftwareConsortium";
+    repo = "ITKSimpleITKFilters";
+    rev = "bb896868fc6480835495d0da4356d5db009592a6";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/itk/generic.nix:27]: How is that commit determined? Seems to be a general thing. Also we skills nice this variable into a let in to not accidentally turn it into a bash variable

Comment (bcdarwin) [reply pkgs/development/libraries/itk/generic.nix:24]: I've lifted these sources into a `let`.

The commits come from the cmake files (but they seemingly aren't too sensitive e.g. both versions of ITK are currently sharing the same versions of these remote modules).  I opened an issue for this [here](https://github.com/NixOS/nixpkgs/issues/222929).

File: pkgs/development/libraries/simpleitk/default.nix (hunk 2 above)
Comment (bcdarwin) [reply pkgs/development/libraries/simpleitk/default.nix:8]: CMake builds rely on package configuration (`.cmake`) files in their dependencies, and with a `dev` output the resulting SimpleITK package configuration file tells depending packages to expect shared libraries in the `/nix/store/xxxxx-simpleitk-dev`, which is wrong.
